### ISSUE TITLE
Unique cargo of the same commodity type is only shown once

### ISSUE THREAD

**Issue by RemainNA** [EDCD/EDMarketConnector#817]
**Please check the [Known Issues](https://github.com/EDCD/EDMarketConnector/issues/618) in case this has already been reported.**

**Please also check if the issue is covered in our [Troubleshooting Guide](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting).** It might be something with a known work around, or where a third party (such as EDSM) is causing logging that is harmless.

*It does not appear to be in either*

**Please complete the following information:**
 - Version 4.1.5
 - OS: Windows 10
 - OS Locale: English
 - Please attach **BOTH** log files as follows:
     1. `%TEMP%\EDMarketConnector.log` from *immediately* after the bug occurs (re-running the application overwrites this file).  [EDMarketConnector.log](https://github.com/EDCD/EDMarketConnector/files/5745960/EDMarketConnector.log)
     1. %TEMP%\EDMarketConnector\EDMarketConnector-debug.log`.  See [Debug Log File](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#debug-log-files).  [EDMarketConnector-debug.log](https://github.com/EDCD/EDMarketConnector/files/5745981/EDMarketConnector-debug.log)

**Describe the bug**
When carrying unique cargo for multiple missions of the same commodity type (i.e. two missions require delivering Food Cartridges) only one of these will be reported.

**To Reproduce**
1. Accept a mission requiring delivering unique cargo, pick up the cargo
2. Accept a second mission requiring the same type of unique cargo, pick it up too
3. Only one of these groups of cargo will be reported in EDMC

**Expected behavior**
Both unique cargo are recorded, either in bulk or as separate instances

**Screenshots**
![EDMC Bug](https://user-images.githubusercontent.com/35419061/103192312-72e6fe00-4895-11eb-921d-b9a8385fdbf4.png)
![EDMC Bug Transactions](https://user-images.githubusercontent.com/35419061/103192311-71b5d100-4895-11eb-9a60-05ab96a14726.png)
![EDMC Bug Inventory](https://user-images.githubusercontent.com/35419061/103192313-737f9480-4895-11eb-9a61-3dde60f34be2.png)

**Additional context**
This was found while using the Cargo Manifest plugin, which I modified to output the cargo dictionary in to the log files.  I did restart between noticing the bug and reporting it, but the same behaviour was observed before the restart, and in another case with different missions/cargo.

**Comment by Athanasius** [comment]
This is a bug in a third-party plugin, not in core EDMC code.  Please report it over on the EDMC Cargo Manifest GitHub -> https://github.com/RemainNA/cargo-manifest

Without looking at their code I can only guess that the plugin looks only at the strict "what cargo we have" plus commodity sell/buy events.  It could potentially start paying attention to the `CargoDepot` events as well.

**Comment by RemainNA** [comment]
Thank you for the suggestion, Cargo Manifest was indeed not checking for CargoDepot events.  However I do still think there may be some issues with EDMC core code based on some additional testing and logging I did.  Cargo Manifest mostly mirrors state['Cargo'], I checked and that isn't reporting the unique cargo correctly either.  I added a little bit of code just to listen to CargoDepot events and when one came through with duplicate unique cargo the state['Cargo'] dictionary changed and only reflected one of the batches of unique cargo.

This is the log directly after picking up the cargo, with Cargo Manifest logging both state['Cargo'] and its internal list this.cargoDict, as well as all CargoDepot events.
[EDMarketConnector.log](https://github.com/EDCD/EDMarketConnector/files/5748860/EDMarketConnector.log)

This is the Elite Cargo.json file (converted to txt for Github) taken at the same time to show how it differs from state['Cargo']
[Cargo.txt](https://github.com/EDCD/EDMarketConnector/files/5748921/Cargo.txt)

It looks like when there are multiple instances of cargo with the same name in Cargo.json one overrides the others, instead of their values being added together.  Sorry if there's anything I'm not understanding.
